web/source/keymanweb.js

```diff
@@ -3705,46 +3704,59 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
+     *                                      isKMWDisabled()] is required.
      */
-    keymanweb._SetupDocument = function(Pelem) { // I1961
-      /**
-       * Function     LiTmp
-       * Scope        Private
-       * @param       {string}    _colon    type of element
-       * @return      {Array}               array of elements of specified type                       
-       * Description  Local function to get list of editable controls
-       */    
-      var LiTmp = function(_colon){
-        return Pelem.getElementsByTagName(_colon);
-      };
+    keymanweb._GetDocumentEditables = function(Pelem) {
+      var possibleInputs = [];
 
-      var possibleInputs = [].concat(
-        util.arrayFromNodeList(LiTmp('INPUT')),
-        util.arrayFromNodeList(LiTmp('TEXTAREA')),
-        util.arrayFromNodeList(LiTmp('IFRAME'))  // Note that isKMWInput() will block these for touch-based devices.
-      );
-      
-      // These are unfortunately accessed via iterator, not array.  Also blocked for touch by isKMWInput().
-      var Lce = util.arrayFromNodeList(document.evaluate 
-          ? document.evaluate('//*[@contenteditable and @contenteditable != "false"]', document, null, XPathResult.ANY_TYPE, null) 
-          : null) // I2457 - support contentEditable elements in mozilla, webkit
-
-      if(Lce) { // I2457 - support contentEditable elements in mozilla, webkit
-        if("iterateNext" in Lce) { // Is sometimes not on mobile solutions.
-          for (var Lc = Lce.iterateNext(); Lc; Lc = Lce.iterateNext()) {
-            possibleInputs.push(Lc);
-          }
-        } else if(Array.isArray(Lce)) { // happens in Chrome mobile emulation.
-          possibleInputs.concat(Lce);
+      if(Pelem.tagName) {
+        var tagName = Pelem.tagName.toLowerCase();
+        if(tagName == 'input' || tagName == 'textarea' || tagName == 'iframe') {
+          possibleInputs.concat(Pelem);
         }
+      } else if(Pelem.nodeName == "#text") {
+        return [];
+      }
+
+      // Constructing it like this also allows for individual element filtering for the auto-attach MutationObserver without errors.
+      if(Pelem.getElementsByTagName) {
+        /**
+         * Function     LiTmp
+         * Scope        Private
+         * @param       {string}    _colon    type of element
+         * @return      {Array<Element>}  array of elements of specified type                       
+         * Description  Local function to get list of editable controls
+         */    
+        var LiTmp = function(_colon){
+          return util.arrayFromNodeList(Pelem.getElementsByTagName(_colon));
+        };
+
+        // Note that isKMWInput() will block IFRAME elements as necessary for touch-based devices.
+        possibleInputs = possibleInputs.concat(LiTmp('INPUT'), LiTmp('TEXTAREA'), LiTmp('IFRAME'));
+      }
+      
+      possibleInputs = possibleInputs.concat(util.arrayFromNodeList(Pelem.querySelectorAll('[contenteditable]')));
```

Comment:
The solution may have issues with IE 8... but then again, I don't think it ever supported `contenteditable`s anyway.  It also doesn't like MutationObservers, so this is not exactly a big concern for us, most likely.